src/Microsoft.Diagnostics.ExtensionCommands/ThreadPoolCommand.cs

```diff
@@ -0,0 +1,242 @@
+                output.WriteRow("Workers Running:", threadPool.ActiveWorkerThreads);
+                output.WriteRow("Workers Idle:", threadPool.IdleWorkerThreads);
+                output.WriteRow("Worker Min Limit:", threadPool.MinThreads);
+                output.WriteRow("Worker Max Limit:", threadPool.MaxThreads);
+                Console.WriteLine();
+
+                ClrType threadPoolType = Runtime.BaseClassLibrary.GetTypeByName("System.Threading.ThreadPool");
+                ClrStaticField usePortableIOField = threadPoolType?.GetStaticFieldByName("UsePortableThreadPoolForIO");
+
+                // Desktop CLR work items.
+                if (PrintWorkItems)
+                {
+                    LegacyThreadPoolWorkRequest[] requests = threadPool.EnumerateLegacyWorkRequests().ToArray();
+                    if (requests.Length > 0)
+                    {
+                        Console.WriteLine($"Work Request in Queue: {requests.Length:n0}");
+                        foreach (LegacyThreadPoolWorkRequest request in requests)
+                        {
+                            Console.CancellationToken.ThrowIfCancellationRequested();
+
+                            if (request.IsAsyncTimerCallback)
+                            {
+                                Console.WriteLine($"    AsyncTimerCallbackCompletion TimerInfo@{request.Context:x}");
+                            }
+                            else
+                            {
+                                Console.WriteLine($"    Unknown Function: {request.Function:x}  Context: {request.Context:x}");
+                            }
+                        }
+                    }
+                }
+
+                // We will assume that if UsePortableThreadPoolForIO field is deleted from ThreadPool then we are always
+                // using C# version.
+                bool usingPortableCompletionPorts = threadPool.Portable && (usePortableIOField is null || usePortableIOField.Read<bool>(usePortableIOField.Type.Module.AppDomain));
+                if (!usingPortableCompletionPorts)
+                {
+                    output = new(Console, (17, ""));
+                    output.WriteRow("Completion Total:", threadPool.TotalCompletionPorts);
+                    output.WriteRow("Completion Free:", threadPool.FreeCompletionPorts);
+                    output.WriteRow("Completion MaxFree:", threadPool.MaxFreeCompletionPorts);
+                    output.WriteRow("Completion Current Limit:", threadPool.CompletionPortCurrentLimit);
+                    output.WriteRow("Completion Min Limit:", threadPool.MinCompletionPorts);
+                    output.WriteRow("Completion Max Limit:", threadPool.MaxCompletionPorts);
+                    Console.WriteLine();
+                }
+
+                if (PrintHillClimbingLog)
+                {
+                    HillClimbingLogEntry[] hcl = threadPool.EnumerateHillClimbingLog().ToArray();
```

Comment:
It looks like the hill climbing log is gotten from the portable implementation. Does it somehow still work with the native implementation / .NET Framework?

Reply:
Ah!  That's right, I forgot the log exists on desktop.  Yes I will add that.  That code is contained in the ClrMD repo though, and the code here will light up when it's built (no changes in this repo needed).

I will merge this PR and add you as a reviewer on the ClrMD PR for the native hillclimbing log.